packages/network/__tests__/fetch-test.ts

```diff
@@ -245,4 +252,44 @@ describe('@data-eden/fetch', function () {
       }
     `);
   });
+
+  test('should be able to introspect on the response as a middleware', async () => {
+    expect.assertions(4);
+
+    async function analyticsMiddleware(
+      request: Request,
+      next: (request: Request) => Promise<Response>
+    ): Promise<Response> {
+      const response = await next(request);
+
+      expect(response.headers).toMatchInlineSnapshot(`
+        Headers {
+          Symbol(map): {
+            "content-type": [
+              "application/json",
+            ],
+            "x-call-id": [
+              "1234567",
+            ],
+            "x-powered-by": [
+              "msw",
+            ],
+          },
+        }
+      `);
+      expect(response.status).toEqual(200);
+
+      return response;
+    }
+
+    const fetch = buildFetch([analyticsMiddleware]);
+
+    const response = await fetch('http://www.example.com/analytics');
```

Comment:
The idea is that clients would specify a request header that the middleware would then consume (e.g. transform to a different one that is sent in the actual network request)

So clients could set a header

X-Feature-Context: A

and the middleware would detect that (in reality the middleware would provide an API for users to set it) and then potentially do something different in the actual network request

Reply:
Although that's really https://github.com/data-eden/data-eden/issues/26